File: modules/x-engine-native-tabbar/android/native-tabbar/src/main/java/com/zkty/nativ/tabbar/Nativetabbar.java
package com.zkty.nativ.tabbar;

import android.util.Log;

import com.tencent.mmkv.MMKV;
import com.zkty.nativ.core.NativeModule;
import com.zkty.nativ.core.XEngineApplication;

import java.io.File;

public class Nativetabbar extends NativeModule implements Itabbar {

    @Override
    public String moduleId() {
        return "com.zkty.native.tabbar";
    }

    @Override
    public void afterAllNativeModuleInited() {

    }
}

File: modules/x-engine-native-tabbar/iOS/Class/Native_tabbar.m
//
//  Native_tabbar.m
//  tabbar
//


#import "Native_tabbar.h"
#import "XENativeContext.h"
#import "iTabbarDelegate.h"

@interface Native_tabbar()
@property (nonatomic, strong) id<iTabbarDelegate> delegate;

@end

@implementation Native_tabbar
NATIVE_MODULE(Native_tabbar)

 - (NSString*) moduleId{
    return @"com.zkty.native.tabbar";
}

- (int) order{
    return 0;
}

- (void)afterAllNativeModuleInited{
} 
 
- (UIViewController*)getCurrentTabItemVC {
    return [self.delegate getCurrentTabItemVC];
}

- (void)registerDelegate:(id)delegate {
    self.delegate=delegate;
}

@end
 
